core/src/avm2/globals/flash/display/bitmap_data.rs

```diff
@@ -440,6 +440,79 @@ pub fn set_pixels<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements `BitmapData.setVector`.
+pub fn set_vector<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let rectangle = args.get_object(activation, 0, "rect")?;
+    // Note - flash player misspells this as 'imputVector'.
+    let vec = args.get_object(activation, 1, "imputVector")?;
+    if let Some(bitmap_data) = this.and_then(|t| t.as_bitmap_data()) {
+        let x = rectangle
+            .get_public_property("x", activation)?
+            .coerce_to_number(activation)?;
+        let y = rectangle
+            .get_public_property("y", activation)?
+            .coerce_to_number(activation)?;
+        let width = rectangle
+            .get_public_property("width", activation)?
+            .coerce_to_number(activation)?;
+        let height = rectangle
+            .get_public_property("height", activation)?
+            .coerce_to_number(activation)?;
+
+        // Clamp to bitmap rect.
+        let bitmap_width = f64::from(bitmap_data.width());
+        let bitmap_height = f64::from(bitmap_data.height());
+        let x_min = x.clamp(0.0, bitmap_width);
```

Comment:
How does the clamping affect the iter? If it tried to write to a 2x2 square starting from `0, -1`, does it start writing index 0 at 0,0 or does it write index 2 at 0,0?

Also maybe use PixelRegion to do the clamping

Reply:
There's an existing test for this - it always starts writing index 0.

Reply:
The initial clamping needs to be done with floating-point values in order for the tests to pass.

Reply:
Awesome, thanks!